src/Vault.sol

```diff
@@ -292,10 +357,24 @@ contract Vault is ERC4626, ERC20Permit, ILiquidationSource, Ownable {
     require(msg.sender == address(_liquidationPair), "Vault/caller-not-LP");
     require(_tokenIn == address(_prizePool.prizeToken()), "Vault/tokenIn-not-prizeToken");
     require(_tokenOut == address(this), "Vault/tokenOut-not-vaultShare");
+    require(_amountOut != 0, "Vault/amountOut-not-zero");
 
-    require(availableBalanceOf(_tokenOut) >= _amountOut, "Vault/amount-gt-available-yield");
+    uint256 _liquidableYield = _availableBalanceOf(_tokenOut);
+    require(_liquidableYield >= _amountOut, "Vault/amount-gt-available-yield");
 
     _prizePool.contributePrizeTokens(address(this), _amountIn);
+
+    if (_yieldFeeRecipient != address(0) && _yieldFeePercentage != 0) {
+      // We get the percentage of liquidated yield and mint the equivalent amount in fees.
+      _mint(
+        _yieldFeeRecipient,
+        _getYieldFeeAmount(
```

Comment:
**Reducing Operations for Better Accuracy**

This logic includes a lot of math operations! Let's reduce the number of operations to improve the floating point accuracy. We know that the `amountOut` is **after** the fee is taken, because of the `availableBalanceOf` fxn. To calculate the fee that should be taken as part of this amount, we can calculate the total:

We know that:

```
amountOut = total * (1 - feePercentage)
```

So to calculate `total` we can go

```
total = amountOut / (1-feePercentage)
```

The fee is the total less the amountOut:

```
fee = total - amountOut
```

So finally we can calculate the fee like so:

```
fee = amountOut / (1-feePercentage) - amountOut
```

Once we bake in the `FEE_PRECISION`, we see the code looks like this:

```
uint256 fee = (_amountOut * FEE_PRECISION) / (FEE_PRECISION - _yieldFeePercentage) - _amountOut
```

This equation uses far fewer operations!

Reply:
Fixed in the following commits:
- https://github.com/pooltogether/v5-vault/pull/6/commits/9ca0d26ae26d72d9cfc4251ce77b2c4368078dc1
- https://github.com/pooltogether/v5-vault/pull/6/commits/ed841e51f11845b7879963c92bf5052dee7153f6